### ISSUE TITLE
Enable utf8mb4

### ISSUE THREAD

**Issue by FriedCircuits** [nextcloud/vm#97]
I have been having issues with the news app and utf8. The fix is to change the db/tables to support utf8mb4. I was thinking this should be setup in the VM for users. I noticed that the last update to NC forced the DB back to UTF8. I found this commit that shows you can enable it for Nextcloud. Maybe this should enabled in the VM?

https://github.com/nextcloud/server/pull/1773
https://github.com/nextcloud/server/pull/1773/commits/cc28f82b369c2e8ebf2d0b4390379b9cda4af40b

Thoughts?

**Comment by enoch85** [comment]
Thanks for your issue. It doesn't seem that it's possible to install MySQL with utf8mb4 directly, in that case it would be an easy fix.

To solve this issue it seems like we need to convert all the tables and columms to utf8mb4. https://mathiasbynens.be/notes/mysql-utf8mb4

Here is another soulution: http://dba.stackexchange.com/a/104866

Feel free to test this and make a PR. Thanks!

**Comment by FriedCircuits** [comment]
That is exactly what I did was to convert them, only the 3 news tables. It looks like with that option enabled NC can create the tables in the first place with utf8mb4.

So what are you thinking, setup the NC/mysqlf config in the VM to support the option?

**Comment by enoch85** [comment]
@FriedCircuits This is not a priority as it doesn't affect all users, but this would be a very good starter PR for you to begin contribute to NC. ;)

**Comment by enoch85** [comment]
Some more info: https://docs.nextcloud.com/server/11/admin_manual/maintenance/mysql_4byte_support.html

**Comment by enoch85** [comment]
Ok, so after some more investigation we will not implement this until it's considered a stable feature in Nextcloud 11.

Something like this would work for someone who wants to implement it:

```
mysqladmin -u root -e "ALTER DATABASE nextcloud_db CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci;"
sudo -u www-data $NCPATH/occ config:system:set mysql.utf8mb4 --type boolean --value="true"
sudo -u www-data $NCPATH/occ maintenance:repair
```

PRs are welcome.

**Comment by enoch85** [comment]
@nerzhul According to the [Nextcloud docs](https://docs.nextcloud.com/server/11/admin_manual/maintenance/mysql_4byte_support.html) it should be like that. Correct me if I'm wrong.

**Comment by enoch85** [comment]
@FriedCircuits This is now in master and will be released with Nextcloud 11.0.2.
